obsidiantools/md_utils.py

```diff
@@ -187,12 +190,18 @@ def _get_ascii_plaintext_from_html(html):
 
 def _get_ascii_plaintext_from_md_file(filepath):
     """md file -> html -> ASCII plaintext"""
-    html = _get_html_from_md_file(filepath)
     # strip out front matter (if any):
-    html = _remove_front_matter(html)
+    html = _get_html_from_md_file(filepath, remove_front_matter=True)
     return _get_ascii_plaintext_from_html(html)
 
 
+def _remove_front_matter_md(md):
+    """Remove front matter from a markdown string."""
+    # remove front matter:
+    front_matter_regex = r'^---\n(.*?)\n---\n'
```

Comment:
It might be a bit safer to use a dedicated library for this purpose (IE [python-frontmatter](https://pypi.org/project/python-frontmatter/)).

The current regex won't match empty frontmatter for example:

```
➔ cat example.py
import re

md = "---\n---\n\nBody content\n"
front_matter_regex = r"^---\n(.*?)\n---\n"

print(re.sub(front_matter_regex, "", md, flags=re.DOTALL))

➔  python example.py
---
---

Body content
```

Reply:
As other quick test cases extending on that, these are also notes that would be front-matter only in Obsidian:

```yml
---



---
```



```yml
---




test: 'a'
---
```

```yml
---


test: 'a'

---
```

Added in lots of new lines essentially, so the regex will need to account for an arbitrary number of new lines between the two rules.

Also haven't had any invalid YAML in my vaults, but this example of invalid YAML could also be interesting to check - in Obsidian this is also a front-matter only note but in the preview it says 'metadata (invalid yaml)':
```yml
---


test: 'a'

___

---
```